File: model/src/main/java/io/github/runtimemodels/chazm/entity/AgentFactory.java
package io.github.runtimemodels.chazm.entity;

import io.github.runtimemodels.chazm.entity.Agent.ContactInfo;
import io.github.runtimemodels.chazm.id.UniqueId;

/**
 * The {@linkplain AgentFactory} interface defines the APIs for constructing {@linkplain Agent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface AgentFactory {

    /**
     * Constructs an {@linkplain Agent}.
     *
     * @param id          the {@linkplain UniqueId} that represents the {@linkplain Agent}.
     * @param contactInfo the {@linkplain ContactInfo} for the {@linkplain Agent}.
     * @return an {@linkplain Agent}.
     */
    Agent buildAgent(UniqueId<Agent> id, ContactInfo contactInfo);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/entity/AgentFactory.kt
package runtimemodels.chazm.model.entity

import runtimemodels.chazm.api.entity.Agent
import runtimemodels.chazm.api.id.UniqueId

/**
 * The [AgentFactory] interface defines the APIs for constructing [Agent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface AgentFactory {
    /**
     * Constructs an [Agent].
     *
     * @param id          the [UniqueId] that represents this [Agent].
     * @param contactInfo a [Map] with contact information for this [Agent].
     * @return an [Agent].
     */
    fun buildAgent(id: UniqueId<Agent>, contactInfo: Map<Any, Any>): Agent
}
